romancal/lib/suffix.py

```diff
@@ -0,0 +1,196 @@
+"""Suffix manipulation
+
+API
+---
+`remove_suffix`
+    Main function used to remove suffixes from
+    file base names.
+
+Notes
+-----
+
+`KNOW_SUFFIXES` is the list used by `remove_suffix`. This
+list is generated by the function `combine_suffixes`. The function uses
+`SUFFIXES_TO_ADD` to add other suffixes that it would otherwise not
+discover or should absolutely be in the list. The function uses
+'SUFFIXES_TO_DISCARD` for strings found that are not to be considered
+suffixes.
+
+Hence, to update `KNOW_SUFFIXES`, update both `SUFFIXES_TO_ADD` and
+`SUFFIXES_TO_DISCARD` as necessary, then use the output of
+`find_suffixes`.
+"""
+import itertools
+import logging
+import re
+
+__all__ = ['remove_suffix']
+
+# Configure logging
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+# Suffixes that are hard-coded or otherwise
+# have to exist. Used by `find_suffixes` to
+# add to the result it produces.
+SUFFIXES_TO_ADD = [
+    'cal', 'crf',
+    'dark',
+    'median',
+    'phot',
+    'ramp', 'rate',
+    'uncal',
+    'dq_init',
+    'assign_wcs',
+    'linearity',
+    'jump',
+    'rampfit',
+    'dark_current'
+]
+
+# Suffixes that are discovered but should not be considered.
+# Used by `find_suffixes` to remove undesired values it has found.
+SUFFIXES_TO_DISCARD = ['pipeline', 'step']
+
+
+# Calculated suffixes.
+# This is produced by the `find_suffixes` function below
+_calculated_suffixes = set([
```

Comment:
These are the default suffixes that are supplied by stpipe that are used if a suffix is not supplied.

Reply:
maybe the comment '# This is produced by the `find_suffixes` function below' is misleading then, because that implies that the list below should be exactly what 'find_suffixes' generates?

Reply:
I couldn't quite determine what JWST was trying to say with that comment so I removed it.